### PR TITLE
fix(trackBy): fix rule when ngFor statements include let assignments

### PR DESCRIPTION
**Pull request by alisd23**
- No longer split by "=" as this broke `*ngFor` statements with `let i = index` statements or similar.
- Add test to cover cases like this.
- Fix line length lint error (maybe only on windows).

### CODE REVIEW

**Comment by alisd23** [comment]
I also upgraded node-sass on my side because `npm install` would not work for me with that version as it couldn't find the binding - I think it doesn't exist for modern versions of node (but I'm not 100% sure of the exact issue). Let me know if you want me to revert that upgrade, and I can just use the CI to see if the tests are passing.

File: src/trackByFunctionRule.ts [hunk 1]
```diff
@@ -24,28 +24,26 @@ export class Rule extends Lint.Rules.AbstractRule {
   }
 }
 
+const ngForExpressionRe = new RegExp(/\*ngFor="(.+)"/);
```
Comment (mgechev) [inline src/trackByFunctionRule.ts]: `/\*ngFor\s*=\s*'|"(.+)'|"/` this might be more accurate it will catch`*ngFor = 'foobar'` as well.

**Comment by mgechev** [comment]
@alisd23 thanks for the PR! Left one comment regarding the regexp.

File: src/trackByFunctionRule.ts (hunk 1 above)
Comment (alisd23) [reply src/trackByFunctionRule.ts]: @mgechev Yep, good point. Will add that in. Also did you see my comment above about node-sass?

Comment (mgechev) [reply src/trackByFunctionRule.ts]: `node-sass` is fine, we use it only in the tests.

Comment (mgechev) [reply src/trackByFunctionRule.ts]: Would you also add a test case with different quotes and whitespace?

Comment (alisd23) [reply src/trackByFunctionRule.ts]: Ok, added a couple more. Should cover all the cases in the Regex.
